pages/playground.js

```diff
@@ -1,140 +1,271 @@
-        />
-        <textarea
-          placeholder={`{           
-            name: "user name",
-            type: "personal",
-            bio: "about the user",
-            links: [
-              {
-                name: "Follow on Twitter",
-                url: "https://twitter.com/username",
-                icon: "FaTwitter",
-              },
-            ],
- }`}
-          className="mt-4 h-80 border-2 hover:border-orange-600 transition-all duration-250 ease-linear rounded px-6 py-2 mb-4 block w-full"
-          name="profileJson"
-          value={profileJson}
-          onChange={(e) => {
-            setProfileJson(e.target.value);
-            setFormatComplete(false);
-            setValidateComplete(false);
-            setSuccessMsg("");
-            setErrMsg("");
-          }}
-        />
-        <div className="flex flex-row justify-end mb-3 gap-2">
-          {!formatComplete && (
-            <Button text="Format" onClick={handleFormatJson} primary={false} />
-          )}
-          {formatComplete && !validateComplete && (
-            <Button
-              text="Validate"
-              onClick={handleValidateJson}
-              primary={false}
-            />
-          )}
-          {formatComplete && validateComplete && (
-            <Button text="Preview" onClick={handlePreview} primary={true} />
-          )}
-        </div>
-
-        {previewModalData && previewModalState && (
-          <PreviewModal
-            toggle={() => setPreviewModalState(!previewModalState)}
-            data={previewModalData}
-          />
-        )}
-      </Page>
-    </>
-  );
+ const [profileJson, setProfileJson] = useState("");
+ const [eventJson, setEventJson] = useState("");
+ const [validateComplete, setValidateComplete] = useState(false);
+ const [validateEventComplete, setValidateEventComplete] = useState(false);
+ const [formatComplete, setFormatComplete] = useState(false);
+ const [formatEventComplete, setFormatEventComplete] = useState(false);
+ const [errorMessage, setErrMsg] = useState("");
+ const [errorEventMessage, setEventErrMsg] = useState("");
+ const [successMessage, setSuccessMsg] = useState("");
+ const [successEventMessage, setSuccessEventMsg] = useState("");
+ const [gitUsername, setGitUsername] = useState("");
+ const [previewModalState, setPreviewModalState] = useState(false);
+ const [previewEventModalState, setPreviewEventModalState] = useState(false);
+ const [previewModalData, setPreviewModalData] = useState();
+ const [previewEventModalData, setEventPreviewModalData] = useState();
+
+
+
+
+ const handleValidateJson = () => {
+   try {
+     JSON.parse(profileJson);
+     setSuccessMsg("Valid Json");
+     setErrMsg("");
+     setValidateComplete(true);
+     return true;
+   } catch (err) {
+     setErrMsg(err.toString());
+     setError(true);
+     setSuccessMsg("");
+     return false;
+   }
+ };
+
+
+ const handleValidateEventJson = () => {

```

Comment:
This looks the same as `handleValidateJson`, I think these could be combined into a single function